src/components/layer-controller/LayerController.js

```diff
@@ -99,33 +126,70 @@ export default function LayerController({ imageData, layerId, handleLayerRemove
-      };
-      const handleChannelRemove = () => {
-        dispatch({ type: 'REMOVE_CHANNEL', layerId, payload: { channelId } });
-      };
-      return (
-        <Grid key={`channel-controller-${channelId}`} item style={{ width: '100%' }}>
-          <ChannelController
-            dimName={dimName}
-            selectionIndex={c.selection[0]}
-            visibility={c.visibility}
-            slider={c.slider}
-            color={c.color}
-            channelOptions={channelOptions}
-            colormapOn={Boolean(colormap)}
-            handlePropertyChange={handleChannelPropertyChange}
-            handleChannelRemove={handleChannelRemove}
-          />
-        </Grid>
-      );
-    });
+  let channelControllers = [];
+  if (dimensions.length > 0) {
+    const { values: channelOptions, field: dimName } = dimensions[0];
+    channelControllers = Object.entries(channels).map(
+      // c is an object like { color, selection, slider, visibility }.
+      ([channelId, c]) => {
+        // Change one property of a channel (for now - soon
+        // nested structures allowing for multiple z/t selecitons at once, for example).
+        const handleChannelPropertyChange = (property, value) => {
+          // property is something like "selection" or "slider."
+          // value is the actual change, like { channel: "DAPI" }.
+          dispatch({
+            type: 'CHANGE_SINGLE_CHANNEL_PROPERTY',
+            layerId,
+            payload: {
+              channelId,
+              property,
+              value,
+            },
+          });
+        };
+        const handleChannelRemove = () => {
+          dispatch({ type: 'REMOVE_CHANNEL', layerId, payload: { channelId } });
+        };
+        return (
+          <Grid
+            key={`channel-controller-${channelId}`}
+            item
+            style={{ width: '100%' }}
+          >
+            <ChannelController
+              dimName={dimName}
+              visibility={c.visibility}
+              slider={c.slider}
+              color={c.color}
+              channelOptions={channelOptions}
+              colormapOn={Boolean(colormap)}
+              handlePropertyChange={handleChannelPropertyChange}
+              handleChannelRemove={handleChannelRemove}
+              dimValues={
+                dimensions.filter(dimension => dimension.field === dimName)[0]
+                  .values
+              }
+            />
+          </Grid>
+        );
+      },
+    );
+  }
 
   const classes = useExpansionPanelStyles();
+  const handleGlobalChannelsSelectionChange = ({ selection, event }) => {
+    // This call updates all channel selections with new global selection.
+    dispatch({
+      type: 'CHANGE_GLOBAL_CHANNELS_SELECTION',
+      layerId,
+      payload: {
+        // See https://github.com/hubmapconsortium/vitessce-image-viewer/issues/176 for why
+        // we have to check mouseup.
+        selection, publish: event.type === 'mouseup',
+      },
```

Comment:
I'm thinking this has to do with the selection state not being observed in the channel dropdown as well

Reply:
While debugging, I didn't find this to be the case.  The controller and Vitessce were publishing the correct events and sending the right props, but Viv was having issues with its state.